### PR TITLE
Separate delete participant form from edit participant form

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Separate delete participant form from edit participant form

Before
----------------------------------------
The delete participant action and the load price field action are overloaded onto the edit participant form - in part via the confusing `EventTab` class. This makes the code really hard to follow.

After
----------------------------------------
I separated the delete participant form to have it's own path..

Technical Details
----------------------------------------
Note that
1) the bulk action for delete participant already has a separate class & form (it didn't seem important to combine them as they message slightly differently)
2) the context handling isn't needed - the key thing it does is handled the situation where there is no form to bounce back to when the Participant form has been used to register a new participant in 'standalone' mode - ie when selected on the form - in which case it goes to the newly created contact's participant tab. This doesn't apply to delete
3) requires menu/rebuild I think
4) my main worry atm is I might have missed some links to update - but I checked & report & contact dashboard don't have them & user participant tab & search results work - delete on view tab seems a bit borked

Comments
----------------------------------------

### CODE REVIEW

File: templates/CRM/Event/Form/Participant.tpl [hunk 1]
```diff
@@ -33,22 +33,7 @@
...
-            <div class="crm-content">
-              {ts 1=$additionalParticipant} There are %1 more Participant(s) registered by this participant.{/ts}
-            </div>
-          {/if}
-        </div>
-        {if $additionalParticipant}
-          {$form.delete_participant.html}
-        {/if}
-        {else} {* If action is other than Delete *}
+      {if 1} {* If action is other than Delete *}
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/Participant.tpl:36]: the if can go later - but getting that right in tpls can be error prone so didn't want to do it here

**Comment by eileenmcnaughton** [comment]
@colemanw I'm wondering if you can help here - I have a hold-out problem - the delete button on the view form - I want that to load the delete code more directly.... 

image

**Comment by colemanw** [comment]
@eileenmcnaughton we have a lot of overloaded pages in CiviCRM that embed the form in the page for *really* no good reason. I've been tackling this in various places and I'm glad to see you making this change here. I'd love to get rid of them all.
IMO the most important thing is to get the form *out* of the page, so I consider the lightest-touch refactor to take this structure where a single page handles all 4 actions (view, add, update, delete)
```xml
<item>
     <path>civicrm/admin/reltype</path> <!-- view, add, update, delete -->
     <page_callback>CRM_Admin_Page_RelationshipType</page_callback>
</item>
```
And break it apart into 2:
```xml
<item>
     <path>civicrm/admin/reltype</path> <!-- view -->
     <page_callback>CRM_Admin_Page_RelationshipType</page_callback>
</item>
<item>
     <path>civicrm/admin/reltype/edit</path> <!-- add, update, delete -->
     <page_callback>CRM_Admin_Form_RelationshipType</page_callback>
</item>
```
In the above (real) example, the form itself is still overloaded with multiple actions, but at least it's not embedded inside an overloaded page -- that's the most heinous part.
Splitting the form further into multiple paths for multiple actions I think is a fine thing to do as a follow-up once the dust settles on getting the form out of the page. The nice thing about storing our paths in the xml schema is that we can change and every SearchKit display will pick it up automatically without needing to be updated.
So all that is to say, I'm fine with merging this, but what about getting the form out of the page?

**Comment by eileenmcnaughton** [comment]
@colemanw hmm - from the point of view of making the form less brittle moving this functionality and the overload functionality that loads the fee information by ajax are the priorities. This does the first of those & I would follow up with the second. The reality is that we probably could replace the extracted form with a SearchKit form pretty quickly if we choose - but it will take a while longer to replace the main form which is 'freed up' by getting this functionality out of it.

So I think if the objective is to support php8.x, get rid of smarty notices & make the sending of workflow messages independent of the form layer (my goal) - this moves us in that direction. At the moment we have the need to retrieve consistent values like event id & participant id & that retrieval works differently in the different forms jammed into `Form_Participant` making it super fragile. But once the form is just doing add & edit (& perhaps still that nasty bulk action) a huge part of the main add/edit form can be removed & what is left is less fragile

 I think it does take us forwards on the part that you think is most heinous but is probably a side issue from my point of view.

**Comment by colemanw** [comment]
Well it's progress either way. And now you know what I think about overloading pages with embedded forms 🤮

**Comment by eileenmcnaughton** [comment]
It looks like it is just update & view that go the through the page - 
image
